packages/bpk-theming/src/BpkThemeProvider.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const uniq = (arr) => {
+  const seen = {};
+  return arr.filter(((item) => {
+    if (seen.hasOwnProperty[item]) {
+      return false;
+    }
+    seen[item] = true;
+    return true;
+  }));
+};
+
+const createStyle = (theme, themeAttributes) => {
+  let style = {};
+  const missingThemeAttributes = [];
+  themeAttributes.forEach((attribute) => {
+    if (theme[attribute]) {
+      const cssName = attribute.replace(/([A-Z])/g, variable => `-${variable.toLowerCase()}`);
+      const value = theme[attribute];
+      style[`--bpk-${cssName}`] = value;
+    } else {
+      missingThemeAttributes.push(attribute);
+    }
+  });
+
+  if (missingThemeAttributes.length > 0) {
+    style = {};
+    console.warn(`BpkThemeProvider: To apply theming, the theme prop must include \`${themeAttributes.join(', ')}\` (missing \`${missingThemeAttributes.join(', ')}\`)`); // eslint-disable-line max-len,no-console
+  }
+  return style;
+};
+
+const BpkThemeProvider = (props) => {
+  const { children, theme, themeAttributes } = props;
+  const dedupedThemeAttributes = uniq(themeAttributes);
+  const style = createStyle(theme, dedupedThemeAttributes);
+  return (
+    <div style={style}>
+      {children}
+    </div>
+  );
+};
+
+BpkThemeProvider.propTypes = {
+  children: PropTypes.node.isRequired,
+  theme: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+  themeAttributes: PropTypes.arrayOf(PropTypes.string).isRequired,
```

Comment:
I think we should use a prop validation for this and verifying that it's done correctly

Reply:
I started writing this, but then I was thinking it would add an additional loop to running this code, instead of it all being handled in one loop in the `render` function. What do you think - am I over-optimising?

Reply:
Yeah I think that's over-optimising. Should be fine imo